docs/service-meshes.asciidoc

```diff
@@ -0,0 +1,295 @@
+
+[float]
+[id="{p}-service-mesh-istio-elasticsearch"]
+===== Elasticsearch
+
+[source,yaml,subs="attributes,callouts"]
+----
+apiVersion: elasticsearch.k8s.elastic.co/{eck_crd_version}
+kind: Elasticsearch
+metadata:
+  name: elastic-istio
+spec:
+  version: {version}
+  http:
+    tls: <1>
+      selfSignedCertificate:
+        disabled: true
+  nodeSets:
+  - name: default
+    count: 3
+    config:
+      node.store.allow_mmap: false
+    podTemplate:
+      metadata:
+        annotations:
+          traffic.sidecar.istio.io/excludeOutboundPorts: "9300" <2>
+          traffic.sidecar.istio.io/excludeInboundPorts: "9300"
+----
+
+<1> Disable the default self-signed certificate generated by the operator and allow TLS to be managed by Istio.
+
+<2> Exclude the transport port (port 9300) from being proxied. Currently ECK does not support switching off X-Pack security and TLS for the Elasticserarch transport port. If Istio is allowed to proxy the transport port, the traffic will be encrypted twice and communication between Elasticsearch nodes will be disrupted.
+
+
+If you do not have link:https://istio.io/docs/tasks/security/authentication/auto-mtls/[automatic mutual TLS] enabled, you may need to create a link:https://istio.io/docs/reference/config/networking/destination-rule/[Destination Rule] to allow the operator to communicate with the Elasticsearch cluster. A communication issue between the operator and the managed Elasticsearch cluster can be detected by looking at the operator logs to see if there are any errors reported with the text `503 Service Unavailable`.
+
+[source,sh]
+----
+kubectl logs -f -n elastic-system -c manager statefulset.apps/elastic-operator
+----
+
+If the operator logs indicate a communications problem, create a `DestinationRule` to enable mutual TLS between the operator and the affected Elasticsearch cluster. For example, the following rule enables mutual TLS for a specific Elasticsearch cluster named `elastic-istio` deployed to the `default` namespace.
+
+[source,yaml]
+----
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: elastic-istio
+spec:
+  host: "elastic-istio-es-http.default.svc.local"
+  trafficPolicy:
+    tls:
+      mode: ISTIO_MUTUAL
+----
+
+Refer to the link:https://istio.io/docs/tasks/security/authentication/mtls-migration/[Istio documentation] for more information about other configuration options affecting authentication between services.
+
+
+[float]
+[id="{p}-service-mesh-istio-kibana"]
+===== Kibana
+
+[source,yaml,subs="attributes,callouts"]
+----
+apiVersion: kibana.k8s.elastic.co/{eck_crd_version}
+kind: Kibana
+metadata:
+  name: elastic-istio
+spec:
+  version: {version}
+  count: 1
+  elasticsearchRef:
+    name: elastic-istio
+  http:
+    tls: <1>
+      selfSignedCertificate:
+        disabled: true
+  podTemplate:
+    metadata:
+      annotations:
+        sidecar.istio.io/rewriteAppHTTPProbers: "true" <2>
```

Comment:
We don't need that one for Elasticsearch?

Reply:
Elasticsearch readiness probe uses a script -- which doesn't seem to be affected by the Istio proxy (TBH, I don't quite understand why). 
Kibana and APM Server use HTTP probes that do get intercepted by the proxy so this annotation is required for those two.

Reply:
Ah that may have changed with https://github.com/elastic/cloud-on-k8s/pull/2528

Reply:
Yes. As this document is mostly meant for 1.0.x users, and because the annotation does not do anything if your readiness probe is a script, I thought it would be simpler to keep it like this for the time being.

Reply:
Should we not keep the version for `master` in sync with the current code in `master` and keep this annotation in a backport to the `1.0` branch?

Reply:
> Should we not keep the version for `master` in sync with the current code in `master` and keep this annotation in a backport to the `1.0` branch?

Shall I merge this PR to `1.0` first and then backport it to `master` with the Kibana annotation removed?

Reply:
GitHub doesn't let me change the PR destination so I have removed the Kibana annotation from this PR. I'll add it back in the backport PR for `1.0`.